contracts/dapis/Median.sol

```diff
@@ -21,18 +21,48 @@ contract Median is Sort, Quickselect {
             if (arrayLength % 2 == 1) {
                 return array[arrayLength / 2];
             } else {
-                return
-                    (array[arrayLength / 2 - 1] + array[arrayLength / 2]) / 2;
+                assert(arrayLength != 0);
+                unchecked {
+                    return
+                        average(
+                            array[arrayLength / 2 - 1],
+                            array[arrayLength / 2]
+                        );
+                }
             }
         } else {
             if (arrayLength % 2 == 1) {
                 return array[quickselectK(array, arrayLength / 2)];
             } else {
-                (uint256 mid1, uint256 mid2) = quickselectKPlusOne(
-                    array,
-                    arrayLength / 2 - 1
-                );
-                return (array[mid1] + array[mid2]) / 2;
+                uint256 mid1;
+                uint256 mid2;
+                unchecked {
+                    (mid1, mid2) = quickselectKPlusOne(
+                        array,
+                        arrayLength / 2 - 1
+                    );
+                }
+                return average(array[mid1], array[mid2]);
+            }
+        }
+    }
+
+    /// @notice Averages two signed integers without overflowing
+    /// @param x Integer x
+    /// @param y Integer y
+    /// @return Average of integers x and y
+    function average(int256 x, int256 y) private pure returns (int256) {
+        if (x > 0 != y > 0) {
+            // No risk of overflow if the signs are different, add them and
+            // divide by 2
+            unchecked {
+                return (x + y) / 2;
+            }
+        } else {
+            // There is risk of overflow if the signs are the same, divide by 2
+            // before adding and compensate for the gobbled bit
+            unchecked {
+                return x / 2 + y / 2 + ((x % 2) + (y % 2)) / 2;
```

Comment:
It feels like this can be implemented without any `if`s

Reply:
Can you please add an example to explain how this code is safe to use with `unchecked`?

Reply:
The sum only overflows if the added numbers have the same sign. So we can safely add and divide by 2 if they don't have the same sign. If they do have the same sign, we now have to divide by 2 before adding the numbers so that the sum doesn't overflow. However, doing so gobbles the carry bit of the summation that two odd numbers would cause, which we have to compensate for.
If we check the boundary conditions, say x = 2^255-1, y = 2^255-1 (largest positive signed integers)
x / 2 = 2^254 - 1
y / 2 = 2^254 - 1
((x % 2) + (y % 2)) / 2 = (1 + 1) / 2 = 1
x / 2 + y / 2 + ((x % 2) + (y % 2)) / 2 = 2^255-1, both doesn't overflow and is correct.
Say x = -2^255, y = -2^255 (smallest negative signed integers)
x / 2 = -2^254
y / 2 = -2^254
((x % 2) + (y % 2)) / 2 = (0 + 0) / 2 = 0
x / 2 + y / 2 + ((x % 2) + (y % 2)) / 2 = -2^255 both doesn't underflow and is correct.

Reply:
`if` is removed in https://github.com/api3dao/airnode-protocol-v1/pull/74/commits/00a296d4c9e7896217d14e33dea8bbd5bd1b6bc9